### PR TITLE
correctly parse destructuring in let outside of strict mode

### LINKED ISSUE #2320: Destructuring in let does not parse correctly outside of strict mode
This code should be parsed without any errors
```ts
let [x] = [1]
let {a: y} = {a: 1}
```
Errors went away if `'use strict'` directive is added to the top of the file

### PR DESCRIPTION
**Pull request by vladima**
Fixes #2320

### CODE REVIEW

File: src/compiler/parser.ts [hunk 1]
```diff
@@ -2975,6 +2975,11 @@ module ts {
             return !scanner.hasPrecedingLineBreak() && isIdentifier()
         }
 
+        function netTokenIsIdentifierOrStartOfDestructuringOnTheSameLine() {
```
Comment (JsonFreeman) [inline src/compiler/parser.ts]: next

File: src/compiler/parser.ts [hunk 2]
```diff
@@ -4873,9 +4878,9 @@ module ts {
         }
 
         function isLetDeclaration() {
-            // It is let declaration if in strict mode or next token is identifier on same line.
+            // It is let declaration if in strict mode or next token is identifier\open brace\open curly on same line.
```
Comment (JsonFreeman) [inline src/compiler/parser.ts]: open bracket / open curly

File: src/compiler/parser.ts [hunk 3]
```diff
@@ -2975,6 +2975,11 @@ module ts {
             return !scanner.hasPrecedingLineBreak() && isIdentifier()
         }
 
+        function netTokenIsIdentifierOrStartOfDestructuringOnTheSameLine() {
+            nextToken();
+            return !scanner.hasPrecedingLineBreak() && (isIdentifier() || token === SyntaxKind.OpenBraceToken || token === SyntaxKind.OpenBracketToken)
```
Comment (JsonFreeman) [inline src/compiler/parser.ts]: wrap this line

**Comment by JsonFreeman** [comment]
:+1:

File: src/compiler/parser.ts (hunk 1 above)
Comment (vladima) [inline src/compiler/parser.ts]: done

File: src/compiler/parser.ts (hunk 2 above)
Comment (vladima) [inline src/compiler/parser.ts]: done

**Pull request merged**